source/load-test-with-locust.rst

```diff
@@ -0,0 +1,479 @@
+  DOJOT_URL: "http://1.2.3.4"
+  # Usually it's the dojot address too (if you don't know for sure, keep the same address as dojot)
+  DOJOT_MQTT_HOST: "1.2.3.4"
+  # The default MQTTS port
+  DOJOT_MQTT_PORT: "8883"
+
+
+.. NOTE::
+  We are assuming you are running the master and the slave in the same machine, i.e. in
+  *non-distributed mode*. Later on will be shown how to distribute slaves between multiple machines.
+
+
+Open ``Docker/scripts/generate_certs/docker-compose.yml`` and change the following environment
+variables:
+
+.. code:: yaml
+
+  # The location of your dojot installation
+  DOJOT_URL: "http://1.2.3.4"
+
+
+Generating the certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As said before, the communication from Locust to dojot is secure, so there is the need to create
+certificates.
+
+There are two ways of simulating devices: you can create fake devices (will not show up in
+dojot's GUI) or real devices. In this part of the tutorial, we will create real devices, so you can
+check the sent messages in the GUI.
+
+Before running the script, we need to initialize the Locust master. Inside Locust repository, run:
+
+.. code:: shell
+
+  docker-compose -f Docker/docker-compose-master.yml up
+
+
+After its initialization, run the ``generate_certs`` script docker compose and enter in it:
+
+.. code:: shell
+
+  docker-compose -f Docker/scripts/generate_certs/docker-compose.yml up -d
+  docker-compose -f Docker/scripts/generate_certs/docker-compose.yml exec generate-certs bash
+
+
+Create the devices in dojot:
+
+.. code:: shell
+
+  generate_certs dojot create --devices 100
+
+
+You can now check that the devices are created in dojot.
+
+.. figure:: images/tutorials/load-test-with-locust/img1.png
+  :width: 100%
+  :align: center
+
+  : Some of the devices that ``generate_certs`` created in dojot.
+
+.. figure:: images/tutorials/load-test-with-locust/img2.png
+  :width: 100%
+  :align: center
+
+  : The template used by ``generate_certs`` to create devices.
+
+
+Generate the certificates for them:
+
+.. code:: shell
+
+  generate_certs cert --dojot
+
+
+The certificates are exported to the ``cert`` directory. Now the test can be initialized!
+
+
+Initializing the slaves
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Locust master does nothing by itself. The one who does all the work of sending requests is the
+Locust slave container. Initialize it by running:
+
+.. code:: shell
+
+  docker-compose -f Docker/docker-compose-slave.yml up
+
+
+You should see a message in the Locust master log saying that one slave has connected to him.
+
+
+Running the test
+^^^^^^^^^^^^^^^^
+
+You are all set to begin the test. To run it, you need to access the Locust interface in your
+browser in ``localhost:8089`` (supposing you are running Locust master in ``localhost``).
+
+Type ``100`` in ``Number of users to simulate``, ``10`` in ``Hatch rate`` and click in
+``Start swarming``. This tells Locust to run 100 clients, creating 10 of them per second.
```

Comment:
This tells Locust to run 100 devices, publish 10  messages per second. ?

Reply:
No, the `Hatch rate` is the speed at which Locust will create clients. The number of messages per second depends on an environment variable in the slave docker compose file.